### PR TITLE
Add additional tests for Validation

### PR DESCRIPTION
**Pull request by nickbristow**
Adds robust testing for validation helper functions

### CODE REVIEW

File: phdi/validation/validation.py [hunk 1]
```diff
@@ -188,24 +185,24 @@ def _validate_text(field, node):
...
+                + config_field.get("fieldName")
                 + " does not match regEx: "
-                + field.get("regEx")
+                + config_field.get("regEx")
             ]
         else:
             return []
     else:
-        if text is not None:
+        if text is not None and text != "":
```
Comment (KennethSkylight) [inline phdi/validation/validation.py:205]: I believe if you just do `if not text` that would work

Both empty string and None are falsey

**Review by KennethSkylight — COMMENTED** [review]
Just one question about the conditional check

File: phdi/validation/validation.py (hunk 1 above)
Comment (nickbristow) [reply phdi/validation/validation.py:205]: tests fail when i do that. not sure why

Comment (BradySkylight) [reply phdi/validation/validation.py:205]: I've had a similar issue when testing text fields as well.  This was just recently though.

**Review by BradySkylight — APPROVED** [review]
These tests look much better and cover all the helper functions. Excellent job Nick!  Thanks for tackling this!

**Pull request merged**